src/ruby.rs

```diff
@@ -51,94 +51,319 @@ extern {
 
+
+    /// Convert a null terminated ASCII encoded character pointer to an [`ID`](struct.id.html)
+    ///
+    /// # Safety
+    ///
+    /// `cstr` must be ASCII encoded
+    ///
+    /// # Defined In
+    ///
+    /// <https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1728>
     pub fn rb_intern(cstr: *const c_char) -> ID;
+
+    /// Convert an ASCII encoded character pointer of the given length to an [`ID`](struct.id.html)
+    ///
+    /// [`rb_intern_const`](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1757-L1760)
+    /// is more or less an alias of this.
+    ///
+    /// # Safety
+    ///
+    /// `cstr` must be ASCII encoded
+    ///
+    /// # Defined In
+    ///
+    /// <https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1729>
     pub fn rb_intern2(ptr: *const c_char, len: c_long) -> ID;
+
+    /// Convert a Ruby [`String`](https://ruby-doc.org/core-2.5.1/String.html) [`VALUE`](struct.VALUE.html)
+    /// to an [`ID`](struct.id.html)
+    ///
+    /// # Safety
+    ///
+    /// Unclear what happens if you pass a `VALUE` that is not a Ruby `String`.
+    ///
+    /// # Defined In
+    ///
+    /// <https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1730>
     pub fn rb_intern_str(string: VALUE) -> ID;
 
+    /// Convert a Ruby [`Symbol`](https://ruby-doc.org/core-2.5.1/Symbol.html) [`VALUE`](struct.VALUE.html)
+    /// to an [`ID`](struct.id.html)
+    ///
+    /// [`RB_SYM2ID`](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L381) and
+    /// [`SYM2ID`](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L386) are currently
+    /// aliases for this.
+    ///
+    /// # Safety
+    ///
+    /// Raises a [`rb_eTypeError`](static.rb_eTypeError.html) if `symbol` is not a Ruby `Symbol`.
+    ///
+    /// # Defined In
+    ///
+    /// <https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L375>
     pub fn rb_sym2id(symbol: VALUE) -> ID;
+
+    /// Convert an [`ID`](struct.ID.html) to a Ruby [`Symbol`](https://ruby-doc.org/core-2.5.1/Symbol.html)
+    /// [`VALUE`](struct.VALUE.html)
+    ///
+    /// # Safety
+    ///
+    /// Unclear what happens if the `ID` is not valid.
+    ///
+    /// # Defined In
+    ///
+    /// <https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L376>
     pub fn rb_id2sym(id: ID) -> VALUE;
 
+    /// Define a new Ruby class
+    ///
+    /// NOTE: If the class is already defined and the superclass is the same
+    /// as specified, it will return the already defined class.
+    ///
+    /// # Safety
+    ///
+    /// * Raises [`rb_eTypeError`](static.rb_eTypeError.html) if the constant name is
+    /// already taken, but the constant is not a C class.
+    /// * Raises [`rb_eTypeError`](static.rb_eTypeError.html) if the class is already
+    /// defined but the class can not be reopened because the superclass does not match.
+    /// * Raises [`rb_eArgError`](static.rb_eArgError.html) if the superclass is null
```

Comment:
Does it only special case null or does it do that with, say a string too? In general we should make a policy of whether we need to spell out "types has to match" or just assume that to be understood.